pkg/apis/cr/v1alpha1/types.go

```diff
@@ -253,49 +290,73 @@ const (
 
 // Location
 type Location struct {
-	Type     LocationType `json:"type"`
-	Bucket   string       `json:"bucket"`
-	Endpoint string       `json:"endpoint"`
-	Prefix   string       `json:"prefix"`
-	Region   string       `json:"region"`
+	// Type specifies the kind of object storage that would be used to upload the backup objects. Currently supported values are: "GCS", "S3Compliant", "Azure" and "Kopia".
+	Type LocationType `json:"type"`
+	// Bucket represents the bucket on the object storage where the backup would be uploaded to.
+	Bucket string `json:"bucket"`
+	// Endpoint specifies the endpoint where the object storage is accessible at.
+	Endpoint string `json:"endpoint"`
+	// Prefix is the string that would be prepended into the object path in the bucket where the objects will be uploaded to.
+	Prefix string `json:"prefix"`
+	// Region represents the region where the bucket mentioned above is present in.
+	Region string `json:"region"`
 }
 
 // CredentialType
 type CredentialType string
 
 const (
+	// CredentialTypeKeyPair represents the Credential being used is of Type key pair.
 	CredentialTypeKeyPair CredentialType = "keyPair"
-	CredentialTypeSecret  CredentialType = "secret"
-	CredentialTypeKopia   CredentialType = "kopia"
+	// CredentialTypeSecret represents the Credential being used is of Type Secret.
+	CredentialTypeSecret CredentialType = "secret"
+	// CredentialTypeKopia represents the Credential being used is of Type Kopia.
+	CredentialTypeKopia CredentialType = "kopia"
 )
 
 // Credential
```

Comment:
If this comment is not necessary, let's remove it.

Reply:
I don't think we can just remove it, go lint my fail. So before committing it please make sure go run `make golint `

Reply:
I think the fields are self-explanatory. So, removing the comments.